### ISSUE TITLE
Location Matrix representation

### ISSUE THREAD

**Issue by chevdor** [hay-kot/homebox#393]
### What is the problem you are trying to solve with this feature?

Location are sometimes shelves or compartmented containers.

here:
|   	|   ‎	|   	|
|:-:	|:--:	|---	|
|   	|  X 	|   	|

or: 
|   	|    	|  X	|
|:-:	|:--:	|---	|
|   	|    ‎|   	|

It would be nice to have a way to visualize such a location.

### What is the solution you are proposing?

I could imagine 2 options:
- supporting an image in the location
- having a way to code a location like a cell in Excel. For instance B12 or AA3.

I think the besty way would be to modelize such a locations with an option turning on "compartments" support.
Then entering the number of rows and cols.

Once defined, we can ensure that items do land in valid compartments.

### What alternatives have you considered?

- I tried using markdown tables in the description of a location without success.
- supporting an image for the location and showing this image could help

The markdown option currently does not work as we don't see the table. Here is an example:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/738724/229760034-a9515fde-2f9f-495d-b093-2d808eb6a9ed.png">

### Additional context

There are those:
![image](https://user-images.githubusercontent.com/738724/229758353-9c32a106-faeb-404e-9c0a-94d10ad00ed8.png)

and those:
<img width="465" alt="Screenshot 2023-04-04 at 12 04 03" src="https://user-images.githubusercontent.com/738724/229758732-53479d8d-d1e8-409b-8a58-a36b2e42d5f0.png">

Knonwing that the 10Ohm resistor I am after is in there is already cool, knowing WHERE it even better :)

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by hay-kot** [comment]
I don't think this is something I can realistically add to the UI. I would maybe consider a PR if it was well tested and had a good UI. Thanks for the request!
